### PR TITLE
Collection sharing 

### PR DESCRIPTION
**Pull request by Reettaphant**
Replace capi query field with a backfill field and allow collections to be be dragged into the field.
screen shot 2015-11-11 at 13 56 41

### CODE REVIEW

File: facia-tool/public/js/utils/drag-dispatcher.js [hunk 1]
```diff
@@ -265,7 +265,8 @@ function persist (sourceItem, newItems, sourceContext, sourceGroup, targetContex
             remove = remove ? remover(sourceContext, sourceGroup, id) : undefined;
         }
 
-        if (sourceContext !== targetContext) {
+        if (sourceContext !== targetContext ||
+            (sourceGroup.parentType === 'Collection' && sourceGroup.parent.state.fromBackfill())) {
```
Comment (piuccio) [inline facia-tool/public/js/utils/drag-dispatcher.js]: this should only be true if backfill is of type `collection`, nothing changes if type `apiquery`

File: facia-tool/public/js/widgets/collection-drop.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        this.displayName = ko.observable();
+        this.backfill = params.backfill;
+        this.apiResults = ko.observableArray();
+        this.apiQueryStatus = ko.observable();
+        this.displayName = ko.observable();
+
+        this[apiQuerySym] = debounce(this.requestApiQueryStatus.bind(this), vars.CONST.searchDebounceMs);
+
+        var that = this;
+        this.apiQuery.subscribe(function(newValue) {
```
Comment (piuccio) [inline facia-tool/public/js/widgets/collection-drop.js]: I prefer if this one uses `this.subscribeOn`

File: facia-tool/public/js/widgets/collection-drop.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+                });
+
+                return contentApi.fetchContentByIds(articleIds)
+                    .then(function (response) {
+                        return {content: response.content};
+                    });
+            }
+        })
+        .catch(function(error) {
+            console.log(error);
```
Comment (piuccio) [inline facia-tool/public/js/widgets/collection-drop.js]: Errors should be handle properly. Display an error message in the UI should be fine, just like `performApiQuery`

File: facia-tool/public/js/widgets/collection-drop.js (hunk 2 above)
Comment (uplne) [inline facia-tool/public/js/widgets/collection-drop.js]: Why?

**Comment by crifmulholland** [comment]
this looks pretty awesome!

File: facia-tool/public/js/modules/droppable.js [hunk 4]
```diff
@@ -8,6 +8,7 @@ import dispatch from 'utils/drag-dispatcher';
 import mediator from 'utils/mediator';
 
 var sourceGroup;
+
 const listeners = Object.freeze({
     dragstart: function (element, event) {
```
Comment (uplne) [inline facia-tool/public/js/modules/droppable.js]: Why not arrow functions?

File: facia-tool/public/js/widgets/collection-drop.js [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+        apiQuery = apiQuery.replace(/\s+/g, '');
+        if (apiQuery) {
+            this.apiQuery(apiQuery);
+            this[apiQuerySym](apiQuery)
+            .then((res = {}) => {
+                var results = res.content || [];
+                this.apiResults(results);
+                this.apiQueryStatus(results.length ? 'valid' : 'invalid');
+            })
+            .catch(() => {
```
Comment (uplne) [inline facia-tool/public/js/widgets/collection-drop.js]: Here you use arrow function and in the code above no. Consistency.

**Comment by uplne** [comment]
@piuccio you should rewrite CSS for this project from scratch. It can be smaller, faster and more maintainable. This is a great project for a really nice and clean CSS.

unicorn

File: facia-tool/public/js/widgets/collection-drop.js (hunk 5 above)
Comment (Reettaphant) [inline facia-tool/public/js/widgets/collection-drop.js]: Yes, I will change this.

File: facia-tool/public/js/modules/droppable.js (hunk 4 above)
Comment (Reettaphant) [inline facia-tool/public/js/modules/droppable.js]: Because it seems they are not used elsewhere in this file.

**Comment by uplne** [comment]
:+1: other than comments

**Comment by uplne** [comment]
So what is happening to this?

**Comment by piuccio** [comment]
waiting for facia-scala-client

**Pull request closed without merging**